src/script/common/c_internal.cpp

```diff
@@ -132,34 +132,51 @@ void script_run_callbacks_f(lua_State *L, int nargs,
 	lua_remove(L, error_handler);
 }
 
+std::string get_optional_origin(lua_State *L, const std::string &modname)
+{
+	// L can be NULL if we get called from scripting_game.cpp
+	if (!L)
+		return "";
+
+	if (modname != "")
+		return "[" + modname + "] ";
+
+	lua_Debug ar;
+
+	if (!lua_getstack(L, 2, &ar))
```

Comment:
Argument 2 is specifically used for the builtin functions, which throw warnings about function calls being deprecated. With argument 1 only, it would log the builtin function instead of the actual calling mod line.
Please provide either a full backtrace (which I should've done when touching this recently) or try to locate the first best backtrace line which actually corresponds to a mod or game source line.

Reply:
`minetest.log` should not need any manual adjustments to work correctly. Providing stack 1 and 2 (if available) backtrace ought to be enough for the most common cases. This includes warnings from builtin and those which are called directly in mods.

Reply:
1 seems to give the path for mods and builtin and 
2 gives the same path for builtin and is absent for mods.